### PR TITLE
Rename auto exporters module

### PR DESCRIPTION
**Pull request by trask**
This PR does 3 things:

* Renames `auto-exporters` parent module to `javaagent-exporters`
* Fixes outdated command in `CONTRIBUTING.md`
* Updates the `archivesBaseName` so that jar file names under `build/libs` match their artifact names

### CODE REVIEW

File: gradle/instrumentation.gradle [hunk 1]
```diff
@@ -16,11 +16,6 @@ byteBuddy {
   }
 }
 
-// TODO(anuraaga): This needs to be added before adding publish.gradle, clean up this ordering restraint.
```
Comment (anuraaga) [inline gradle/instrumentation.gradle:19]: Thanks!

**Comment by trask** [comment]
> Just to make sure, you did check `publishToMavenLocal` artifacts?

Yup 👍

File: smoke-tests/smoke-tests.gradle [hunk 2]
```diff
@@ -6,18 +6,18 @@ dependencies {
...
 }
 
 test {
-  dependsOn = [':javaagent:shadowJar', ':auto-exporters:opentelemetry-auto-exporter-logging:shadowJar']
+  dependsOn = [':javaagent:shadowJar', ':javaagent-exporters:logging:shadowJar']
 
   doFirst {
     jvmArgs "-Dio.opentelemetry.smoketest.agent.shadowJar.path=${project(':javaagent').tasks.shadowJar.archivePath}"
-    jvmArgs "-Dotel.exporter.jar=${project(':auto-exporters:opentelemetry-auto-exporter-logging').tasks.shadowJar.archivePath}"
+    jvmArgs "-Dotel.exporter.jar=${project(':javaagent-exporters:logging').tasks.shadowJar.archivePath}"
```
Comment (iNikem) [inline smoke-tests/smoke-tests.gradle:21]: That is now unused, btw. Seems like I forgot to delete. All smoke tests should now use Collector.

File: smoke-tests/smoke-tests.gradle [hunk 3]
```diff
@@ -6,18 +6,17 @@ dependencies {
   api deps.spock
   api project(':testing-common')
 
-  implementation project(':auto-exporters:opentelemetry-auto-exporter-logging')
+  implementation project(':javaagent-exporters:logging')
```
Comment (iNikem) [inline smoke-tests/smoke-tests.gradle:9]: Then this is not needed as well :)

File: smoke-tests/smoke-tests.gradle [hunk 4]
```diff
@@ -6,18 +6,17 @@ dependencies {
...
   implementation deps.slf4j
   implementation deps.opentelemetryProto
   implementation "org.testcontainers:testcontainers:1.14.3"
   implementation 'com.google.protobuf:protobuf-java-util:3.12.4'
 }
 
 test {
-  dependsOn = [':javaagent:shadowJar', ':auto-exporters:opentelemetry-auto-exporter-logging:shadowJar']
+  dependsOn = [':javaagent:shadowJar', ':javaagent-exporters:logging:shadowJar']
```
Comment (iNikem) [inline smoke-tests/smoke-tests.gradle:17]: And this

**Pull request merged**